### PR TITLE
Create StalinSort.java

### PR DESCRIPTION
**Pull request by anant-jain01**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/StalinSort.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        for (int i = 1; i < array.length; i++) {
+            if (SortUtils.less(result[index - 1], array[i])) {
+                result[index++] = array[i];
+            }
+        }
+
+        return trimArray(result, index);
+    }
+
+    private <T> T[] trimArray(T[] array, int length) {
```
Comment (saahil-mahato) [inline src/main/java/com/thealgorithms/sorts/StalinSort.java:34]: I think this is redundant. Is there a reason to use this?

Comment (saahil-mahato) [inline src/main/java/com/thealgorithms/sorts/StalinSort.java:34]: This just adds unnecessary overhead. We can remove this if we just use ArrayList.
```
List<T> result = new ArrayList<>();
        result.add(array[0]); 

        for (int i = 1; i < array.length; i++) {
            if (SortUtils.less(result.get(result.size() - 1), array[i])) {
                result.add(array[i]);
            }
        }
```

File: src/main/java/com/thealgorithms/sorts/StalinSort.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * elements that violate the sort order.
+ * 
+ * For more information, see: 
+ * https://en.wikipedia.org/wiki/Stalin_sort
+ */
+public class StalinSort implements SortAlgorithm {
+
+    @Override
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        if (array.length == 0) {
```
Comment (saahil-mahato) [inline src/main/java/com/thealgorithms/sorts/StalinSort.java:17]: add `null` check here to prevent null pointer expection.

**Comment by saahil-mahato** [comment]
Please add tests.

**Review by siriak — CHANGES_REQUESTED** [review]
Please add JUnit tests

**Pull request closed without merging**